codetf.schema.json

```diff
@@ -237,9 +237,34 @@
             "description": "Whether the finding was fixed by the codemod"
           },
           "reason": {
-            "type": "string",
-            "description": "Reason the finding was not fixed"
```

Comment:
Maybe the failures should be captured at the `result` level, and the successes should be captured at the individual `changeset`. 

Without that, it's not clear which individual code change is associated with which finding (except the user cross correlating the fix locations manually.)

Reply:
Both of these are additive changes, so it doesn't seem burdensome.

Reply:
I'm a little unclear on what you'd like to change here. I think this change adds more detailed failure metadata to the `result` but do you want something else for `changeset`? What are we missing there? More information about the finding itself?

Reply:
I want downstream consumers to be able to:
* easily tell which successful fix is associated with which `$changeset[n].change`
* easily tie a failed fix, when possible, to a file (and optionally line number)
* provide a way to provide failed fix information that can't be associated with a changeset entry 

I'm thinking something like this:

```json
"codemod" : "vendor:python/sql-injection",
"detector" : 
    ...
   "failedFixes" : [ // these are all failures that we can't associate with a changeset entry
      {
         "id" : "5",
         "reason" : "couldn't find the file", // reason is required
         "path" : "src/nonexistent/foo.py"
      },
      {
         "id" : "7",
         "reason" : "would have caused side effects",
         "path" : "src/no_associated_changes.py",
         "line" : "25" // line is optional
      }
   ],
   ...
   "changeset" : [
       "path" : "src/bar.py",
       "diff" : ...
       "changes" : [
         {
            "lineNumber" : "51" // already exists 
            "diffSide" : "LEFT",
            "description" : "Fixed SQLi",
            "detectorFixes" : [ // we add this new array
             {
                "id" : "6" // we just add id, but make it an object so we can add more later
             }
         ]
       ]
   ]
]
```

Successful fix metadata would now be directly associated with a `changeset` entry. Without doing this, the CodeTF has to capture a lot of redundant data in the `result`-level data so that downstream consumers can do storytelling at the correct location.